Instruct how to publish extensions to the registry (#22038)
quarkusio/quarkus: docs/src/main/asciidoc/writing-extensions.adoc

the notion of Ecosystem CI. The Ecosystem CI link:https://github.com/quarkusio/quarkus-ecosystem-ci/blob/main/README.adoc[README]
has all the details on how to set up a GitHub Actions job to take advantage of this capability, while this link:https://www.youtube.com/watch?v=VpbRA1n0hHQ[video] provides an overview
of what the process looks like.

== Publish your extension in registry.quarkus.io

Before publishing your extension to the link:https://quarkus.io/guides/tooling[Quarkus tooling], make sure that the following requirements are met:

* The `quarkus-extension.yaml` file (in the extension's `runtime/` module) has the minimum metadata set:
** `name`

> ** `description` (Otherwise set it in the `runtime/pom.xml`'s `<description>` element)

Author: How about this? 

```suggestion
** `description` (unless you have it already set in the `runtime/pom.xml`'s `<description>` element, which is the recommended approach)
```

> ** `description` (Otherwise set it in the `runtime/pom.xml`'s `<description>` element)

Author: It would be nice if we had a Maven Plugin that would check if all the necessary requirements to publish the extension are met

> ** `description` (Otherwise set it in the `runtime/pom.xml`'s `<description>` element)

Reviewer: Your proposal looks good.

* Your extension is published in Maven central

> Then you must create a pull-request adding a `your_extension.yaml` file in the `extensions/` directory in the link:https://github.com/quarkusio/quarkus-extension-catalog[Quarkus Extension Catalog]. The YAML must have the following structure:

Reviewer: ```suggestion
Then you must create a pull request adding a `your-extension.yaml` file in the `extensions/` directory in the link:https://github.com/quarkusio/quarkus-extension-catalog[Quarkus Extension Catalog]. The YAML must have the following structure:
```
Minor change but it will avoid having people creating files with underscores.

```yaml
group-id: <YOUR_EXTENSION_RUNTIME_GROUP_ID>
artifact-id: <YOUR_EXTENSION_RUNTIME_ARTIFACT_ID>
```

> That's all. Once the pull-request is merged, a scheduled job will check Maven Central for new versions and update the link:extension-registry-user.adoc[Quarkus Extension Registry].

Reviewer: ```suggestion
That's all. Once the pull request is merged, a scheduled job will check Maven Central for new versions and update the link:extension-registry-user.adoc[Quarkus Extension Registry].
```
